src/CustomElementInternals.js

```diff
@@ -9,13 +9,42 @@
  */
 
 import * as Utilities from './Utilities.js';
+import Deferred from './Deferred.js';
 import CEState from './CustomElementState.js';
 
+const PENDING_LAZY_DEFINITION = {};
+
 export default class CustomElementInternals {
-  constructor() {
+  constructor(preferPerformance) {
+    /**
+     * @private
+     * @type {!Map<string, !Deferred<undefined>>}
+     */
+    this._whenDefinedDeferred = new Map();
+
+    /**
+     * The default flush callback triggers the document walk synchronously.
+     * @type {!Function}
+     */
+    this.flushCallback = fn => fn();
```

Comment:
I think the layering would be a bit cleaner if this was private and the actual wrapping part of `polyfillWrapFlushCallback` was moved to a function here (`wrapFlushCallback`?).

Reply:
Done.